server/app/controllers/ti/TrustedIntermediaryController.java

```diff
@@ -160,31 +197,49 @@ public Result addApplicant(Long id, Http.Request request) {
     if (!trustedIntermediaryGroup.get().id.equals(id)) {
       return unauthorized();
     }
-    Form<AddApplicantToTrustedIntermediaryGroupForm> form =
+    Form<TiClientInfoForm> form =
         tiService.addNewClient(
-            formFactory
-                .form(AddApplicantToTrustedIntermediaryGroupForm.class)
-                .bindFromRequest(request),
-            trustedIntermediaryGroup.get());
-    if (!form.hasErrors()) {
-      return redirect(
-              routes.TrustedIntermediaryController.dashboard(
-                  /* nameQuery= */ Optional.empty(),
-                  /* dayQuery= */ Optional.empty(),
-                  /* monthQuery= */ Optional.empty(),
-                  /* yearQuery= */ Optional.empty(),
-                  /* page= */ Optional.of(1)))
-          .flashing(
-              "success",
-              String.format(
-                  "Successfully added new client: %s %s",
-                  form.value().get().getFirstName(), form.value().get().getLastName()));
+            formFactory.form(TiClientInfoForm.class).bindFromRequest(request),
+            trustedIntermediaryGroup.get(),
+            messagesApi.preferred(request));
+    if (form.hasErrors()) {
+      return ok(
+          editTiClientView.render(
+              /* tiGroup= */ trustedIntermediaryGroup.get(),
+              /* personalInfo= */ ApplicantPersonalInfo.ofLoggedInUser(
+                  Representation.builder()
+                      .setName(
+                          form.value().get().getFirstName()
+                              + " "
+                              + form.value().get().getLastName())
+                      .build()),
+              /* request= */ request,
+              /* messages= */ messagesApi.preferred(request),
+              /* accountIdToEdit= */ Optional.empty(),
+              /* applicantIdOfTi= */ civiformProfile
+                  .get()
+                  .getApplicant()
+                  .toCompletableFuture()
+                  .join()
+                  .id,
+              /* tiClientInfoForm= */ Optional.of(form)));
     }
-    return redirectToDashboardWithError(getValidationErrors(form.errors()), form);
+    return redirect(
+            routes.TrustedIntermediaryController.dashboard(
+                /* nameQuery= */ Optional.empty(),
+                /* dayQuery= */ Optional.empty(),
+                /* monthQuery= */ Optional.empty(),
+                /* yearQuery= */ Optional.empty(),
+                /* page= */ Optional.of(1)))
+        .flashing(
+            "success",
+            String.format(
+                "Successfully added new client: %s %s",
+                form.value().get().getFirstName(), form.value().get().getLastName()));
```

Comment:
question: It looks like Melanie's https://github.com/civiform/civiform/pull/6969 PR changes this success behavior, do you expect that Melanie's changes will go in first and then you'll update the behavior here?

Reply:
It a race condition 🐎

Reply:
Most likely she will go first and I will merge my changes.